h/models/group.py

```diff
@@ -229,16 +235,18 @@ def created_by(cls, session, user):
 
 TypeFlags = namedtuple("TypeFlags", "joinable_by readable_by writeable_by")
 
-OPEN_GROUP_TYPE_FLAGS = TypeFlags(
-    joinable_by=None, readable_by=ReadableBy.world, writeable_by=WriteableBy.authority
-)
-
-PRIVATE_GROUP_TYPE_FLAGS = TypeFlags(
-    joinable_by=JoinableBy.authority,
-    readable_by=ReadableBy.members,
-    writeable_by=WriteableBy.members,
-)
-
-RESTRICTED_GROUP_TYPE_FLAGS = TypeFlags(
-    joinable_by=None, readable_by=ReadableBy.world, writeable_by=WriteableBy.members
-)
+GROUP_TYPE_FLAGS = {
+    "open": TypeFlags(
+        joinable_by=None,
+        readable_by=ReadableBy.world,
+        writeable_by=WriteableBy.authority,
+    ),
+    "private": TypeFlags(
+        joinable_by=JoinableBy.authority,
+        readable_by=ReadableBy.members,
+        writeable_by=WriteableBy.members,
+    ),
+    "restricted": TypeFlags(
+        joinable_by=None, readable_by=ReadableBy.world, writeable_by=WriteableBy.members
+    ),
+}
```

Comment:
I rearranged these three separate `*_GROUP_TYPE_FLAGS` constants into a single `GROUP_TYPE_FLAGS` dict, which makes the code nicer in several places.